framework/include/executioners/FEProblemSolve.h

```diff
@@ -51,4 +59,11 @@ class FEProblemSolve : public NonlinearSolveObject
   /// member in this way allows for easy boolean operations, e.g. if (_num_grid_steps)
   /// as opposed to if (_num_grids)
   const unsigned int _num_grid_steps;
+
+  /// Whether we are using fixed point iterations for multi-system
+  const bool _using_multi_sys_fp_iterations;
+  /// Convergence object to assess the convergence of the multi-system fixed point iteration
+  Convergence * _multi_sys_fp_convergence;
+  /// Whether to only solve the first nonlinear system, which is the legacy behavior
+  const bool _solve_only_first_system;
```

Comment:
I don't like this data member. If this can be true, then this class is not truly a `MultiSystemSolveObject`. I think either this data member needs to go away or `FEProblemSolve` should not inherit from `MultiSystemSolveObject`

Reply:
Ok that's fair. I did not need this, until I tried to get the optimization module tests to pass. It uses two solve objects m, one for each system.

I have a different solution you might like more.

Reply:
Removed